openedx/core/djangoapps/config_model_utils/models.py

```diff
@@ -0,0 +1,125 @@
+"""
+Useful ConfigurationModel subclasses
+
+StackedConfigurationModel: A ConfigurationModel that can be overridden at site, org and course levels
+"""
+
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from collections import namedtuple
+
+from django.conf import settings
+from django.db import models
+from django.contrib.sites.models import Site
+from django.utils.translation import ugettext_lazy as _
+
+from config_models.models import ConfigurationModel
+from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class StackedConfigurationModel(ConfigurationModel):
+    """
+    A ConfigurationModel that stacks Global, Site, Org, and Course level
+    configuration values.
+    """
+    class Meta(object):
+        abstract = True
+
+    KEY_FIELDS = ('site', 'org', 'course')
+
+    enabled = models.NullBooleanField(default=None, verbose_name=_("Enabled"))
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, null=True)
+    org = models.CharField(max_length=255, db_index=True, null=True)
+    course = models.ForeignKey(
+        CourseOverview,
+        on_delete=models.DO_NOTHING,
+        null=True,
+    )
+
+    @classmethod
+    def attribute_tuple(cls):
+        "Returns a namedtuple with all attributes that can be overridden on this config model."
```

Comment:
could you provide a string representation of what this return value would look like in the docstring?

Reply:
Yes, although it's going to vary based on `STACKED_FIELDS`. But I should mention that.